### PR TITLE
Remove unnecessary reassignments and unused vars

### PR DESCRIPTION
**Pull request by beeankha**

### CODE REVIEW

**Review by ryanskeith — COMMENTED** [review]
Nice removals.

File: conda_build/conda_interface.py [hunk 1]
```diff
@@ -4,123 +4,93 @@
...
+    CondaError,
+    CondaHTTPError,
+    LinkError,
+    LockError,
+    NoPackagesFoundError,
+    PaddingError,
+    UnsatisfiableError,
+)
+from conda.base.context import (  # noqa: F401
+    non_x86_linux_machines,
```
Comment (kenodegard) [inline conda_build/conda_interface.py:85]: in conda.base.context this is called `non_x86_machines`:

```suggestion
    non_x86_machines as non_x86_linux_machines,
```

File: conda_build/conda_interface.py [hunk 2]
```diff
@@ -4,123 +4,93 @@
...
+    context,
+    get_prefix,
```
Comment (kenodegard) [inline conda_build/conda_interface.py:87]: we need this to be called differently:

```suggestion
    get_prefix as context_get_prefix,
```

File: conda_build/conda_interface.py [hunk 3]
```diff
@@ -131,24 +101,15 @@ def try_exports(module, attr):
 create_default_packages = context.create_default_packages
 
 get_rc_urls = lambda: list(context.channels)
-get_prefix = partial(context_get_prefix, context)
+get_prefix = partial(context_get_prefix, context)  # noqa: F811, F821
```
Comment (kenodegard) [inline conda_build/conda_interface.py:104]: neither noqa is needed with the fix above:

```suggestion
get_prefix = partial(context_get_prefix, context)
```

**Comment by kenodegard** [comment]
Ran a diff between the old `conda_build.conda_interface` locals and the new locals and this is what was removed:

- `_get_version_from_git_tag`
- `arch_name`
- `bits`
- `conda`
- `conda_43`
- `conda_44`
- `conda_45`
- `conda_46`
- `conda_47`
- `conda_48`
- `conda_411`
- `get_local_urls`
- `try_exports`

**Review by kenodegard — APPROVED** [review]

**Comment by beckermr** [comment]
This PR broke the public API. Does conda build have a policy on API breaks? Should we have bumped to 5 because of this change?

File: conda_build/conda_interface.py [hunk 4]
```diff
@@ -4,123 +4,93 @@
...
-    pass
-
-# All of these conda's are older than our minimum dependency
-conda_43 = True
-conda_44 = True
-conda_45 = True
-conda_46 = True
-conda_47 = True
-conda_48 = True
-conda_411 = True
```
Comment (beckermr) [inline conda_build/conda_interface.py:39]: These are public attributes and were removed if I am not mistaken?

@kenodegard

Comment (kenodegard) [reply conda_build/conda_interface.py:39]: @jezdez [responded on Matrix](https://matrix.to/#/!TfaKrPTflzGMvGhxtd:gitter.im/$uPoHHpqtRvcoPFiAT3GbDgkAP-Qtte7jvkUsXjc3ecE?via=gitter.im&via=matrix.org&via=salt-rock-lamp.ems.host), reposting here for visibility:

> Sorry for the mess @beckermr, seems like it wasn't clear that this was used elsewhere, for better or worse. I'll kick off the discussion to apply [CEP 8](https://github.com/conda-incubator/ceps/blob/main/cep-8.md) and [9](https://github.com/conda-incubator/ceps/blob/main/cep-9.md) on conda-build next week to make sure we have some lead-time when removing things like this.

Comment (danpetry) [reply conda_build/conda_interface.py:39]: yeah, this broke the concourse (anaconda CI) command-line tool

**Pull request merged**